src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,145 @@
+import org.apache.mahout.cf.taste.impl.recommender.GenericUserBasedRecommender;
+import org.apache.mahout.cf.taste.impl.similarity.PearsonCorrelationSimilarity;
+import org.apache.mahout.cf.taste.model.DataModel;
+import org.apache.mahout.cf.taste.recommender.RecommendedItem;
+import org.apache.mahout.cf.taste.neighborhood.UserNeighborhood;
+import org.apache.mahout.cf.taste.recommender.UserBasedRecommender;
+import org.apache.mahout.cf.taste.similarity.UserSimilarity;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+
+/**
+ *
+ * @author ifrescas
+ */
+
+public class MovieRecommender {
+    
+    private int totalReviews = 0;
+    private int totalProducts = 0;
+    private int totalUsers = 0;
+
+    //Hash map para almacenar ids
+    //HashMap<String, Integer> productsMap = new HashMap<String, Integer>();
+    //HashMap<String, Integer> usersMap = new HashMap<String, Integer>();
+
+    BiMap<String, Integer> productsMap = HashBiMap.create();
+    BiMap<String, Integer> usersMap = HashBiMap.create();
+
+
+    public MovieRecommender(String path) throws  IOException {
+
+
+
+        //Escritura de datos para crear archivo .csv
+        File result = new File("Result.csv");
+        FileWriter fw = new FileWriter(result);
+        BufferedWriter bw = new BufferedWriter(fw);
+
+        // Lectura de datos
+        String infile = path;
+        GZIPInputStream in = new GZIPInputStream(new FileInputStream(infile));
+        Reader decoder = new InputStreamReader(in);
+        BufferedReader br = new BufferedReader(decoder);
+        String line;
+
+        String cvsProd="";
+        String cvsUser="";
+        String cvsScore = "";
+
+        int prodNum = 0;
+        int userNum = 0;
+        int reviewNum = 0;
+
+
+        while ((line = br.readLine()) != null) {
+            if (line.startsWith("product/productId:")) {
+                reviewNum++;
+
+                String [] parts = line.split(": ");
+                String productId = parts[1];
+                
+                if (!productsMap.containsKey(productId)) {
+                    productsMap.put(productId,prodNum++);
+                }
+                cvsProd = productsMap.get(productId).toString();
+
+            } else if (line.startsWith("review/userId")) {
+                String [] parts = line.split(": ");
+                String userId = parts[1];
+
+                if(!usersMap.containsKey(userId)) {
+                    usersMap.put(userId,userNum++);
+                }
+                cvsUser = usersMap.get(userId).toString();
+
+            } 
+            else if (line.startsWith("review/score")) {
+                String [] parts = line.split(": ");
+                cvsScore = parts[1];
+
+                String entry = String.format("%s,%s,%s\n",   cvsUser ,cvsProd, cvsScore);
+                bw.write(entry);
+            }
+        }
+        
+        br.close();
+        bw.close();
```

Comment:
Not entirely sure on the version of Java used here, but Try with Resources was added to have try/catch clauses without the necessity to manually closing the reader and/or writer. This is a nice to have feature, and is not something that should stop this PR's process.

Reference: [Try with Resources](https://www.baeldung.com/java-try-with-resources)